### PR TITLE
Updating mstelemetry portfile for windows

### PR DESCRIPTION
**Pull request by anaaru**
Got clienttelemetry.lib to build with MSVC v142 platform toolset.

### CODE REVIEW

File: tools/ports/mstelemetry/portfile.cmake [hunk 1]
```diff
@@ -4,45 +4,80 @@ message("CMAKE_CURRENT_SOURCE_DIR=${CMAKE_CURRENT_SOURCE_DIR}")
...
-file(INSTALL ${SOURCE_PATH}/LICENSE DESTINATION ${CURRENT_PACKAGES_DIR}/share/${PORT} RENAME copyright)
+    vcpkg_install_msbuild(
+        SOURCE_PATH ${SOURCE_PATH}
+        PROJECT_SUBPATH Solutions/MSTelemetrySDK.sln
+        LICENSE_SUBPATH LICENSE
+        RELEASE_CONFIGURATION Release
+        DEBUG_CONFIGURATION Debug
+        OPTIONS /p:MATSDK_SHARED_LIB=1
+        PLATFORM ${VCPKG_TARGET_ARCHITECTURE}
+        PLATFORM_TOOLSET v142
```
Comment (maxgolov) [inline tools/ports/mstelemetry/portfile.cmake:73]: I think we can add two options here:
- one is to respect env variable `PLATFORM_TOOLSET` , if it is already set (for those who still use vs2017-vc141)
- another option is to auto-detect the compiler. It's a bit more involved. But I can help with a script for that.

In general, maybe just replacing it with a custom variable logic - if set, use the value from var; else if not set, then assume the tool chain is v142 - would be good enough at first.

File: tools/ports/mstelemetry/portfile.cmake [hunk 2]
```diff
@@ -4,45 +4,80 @@ message("CMAKE_CURRENT_SOURCE_DIR=${CMAKE_CURRENT_SOURCE_DIR}")
 message("CMAKE_MODULE_PATH=${CMAKE_MODULE_PATH}")
 message("CMAKE_CURRENT_LIST_DIR=${CMAKE_CURRENT_LIST_DIR}")
 
-execute_process(COMMAND "${CMAKE_CURRENT_LIST_DIR}/get_repo_name.sh" OUTPUT_VARIABLE REPO_NAME ERROR_QUIET)
+if (UNIX)
+    execute_process(COMMAND "${CMAKE_CURRENT_LIST_DIR}/get_repo_name.sh" OUTPUT_VARIABLE REPO_NAME ERROR_QUIET)
+else()
+    # execute_process(COMMAND git config --get remote.origin.url OUTPUT_VARIABLE REPO_URL ERROR_QUIET)
```
Comment (maxgolov) [inline tools/ports/mstelemetry/portfile.cmake:10]: I can add a batch file `get_repo_name.cmd` here, but we also then need to assume that `git.exe` is also in the path.

**Comment by maxgolov** [comment]
@anaaru - I think it's generally looking good. Let's merge it, and I'll add a few improvements in a separate PR, add you as reviewer.

**Review by maxgolov — APPROVED** [review]

**Comment by maxgolov** [comment]
@anaaru - can you send in a separate follow-up PR that adds a bit of documentation to this file ?
https://github.com/microsoft/cpp_client_telemetry/blob/master/docs/building-with-vcpkg.md
It _should_ work, but maybe there are a few quirks in the process. I think I originally tested it on Linux only.

**Comment by maxgolov** [comment]
@lulululululu - I think this PR should partially address what you asked for in #781

**Review by mkoscumb — APPROVED** [review]

**Pull request merged**